simln-lib/src/lib.rs

```diff
@@ -1,3 +1,6 @@
+#![deny(rustdoc::broken_intra_doc_links)]
+#![deny(missing_docs)]
```

Comment:
If we remove `missing_docs` I believe we should be able to keep `#![deny(rustdoc::broken_intra_doc_links)]` as-is to make sure code references are correct.